tests/testthat/test-fitNBthDE.R

```diff
@@ -151,21 +151,100 @@ test_that("fitNBthDE produces desired results, CTA", {
     sizescalebythreshold = TRUE,
     sizefactrec = FALSE
   )
-
-
+  
+  
   #   sizefact is the same as sizefact_start.
   expect_true("sizefact" %in% names(NBthDEmod3))
   expect_true(is.vector(NBthDEmod3[["sizefact"]]))
   expect_true(length(NBthDEmod3[["sizefact"]]) == length(sizefact_start))
   expect_true(all(NBthDEmod3[["sizefact"]] == sizefact_start))
+  
+  
+})
 
-
+test_that("fitNTthDE works with dgCMatrix format",{
+  data("demoData")
+  demoData <- demoData[, c(1:5, 33:37)]
+  set.seed(413)
+  demoData <- fitPoisBG(demoData, size_scale = "sum")
+  demoData <- aggreprobe(demoData, use = "cor")
+  negdat <- demoData[which(Biobase::fData(demoData)$CodeClass == "Negative"), ]
+  countmat <- Biobase::exprs(negdat)
+  countmat = as(countmat, "dgCMatrix")
+  result <- fitPoisBG(
+    object = countmat,
+    iterations = 10,
+    tol = 1e-3,
+    size_scale = "sum")
+  demoData[["sizefact"]] <- result$sizefact[Biobase::sampleNames(demoData)]
+  Biobase::fData(demoData)[["featfact"]] <- NA
+  Biobase::fData(demoData)[["featfact"]][match(names(result$featfact), Biobase::featureNames(demoData), nomatch = 0)] <- result$featfact
+  # Case 1: adjustment factor 5, no outlier removal, no prior
+  demoData <- BGScoreTest(demoData,
+                          adj = 5,
+                          removeoutlier = FALSE, useprior = FALSE)
+  demoData$slidename <- substr(demoData[["slide name"]], 12, 17)
+  thmean <- 1 * mean(fData(demoData)$featfact, na.rm = TRUE)
+  demo_pos <- demoData[which(!fData(demoData)$CodeClass == "Negative"), ]
+  demo_neg <- demoData[which(fData(demoData)$CodeClass == "Negative"), ]
+  sc1_scores <- fData(demo_pos)[, "scores"]
+  names(sc1_scores) <- fData(demo_pos)[, "TargetName"]
+  features_high <- ((sc1_scores > quantile(sc1_scores, probs = 0.4)) &
+                      (sc1_scores < quantile(sc1_scores, probs = 0.95))) |>
```

Comment:
We do not suggest using pipes in our open source code. Can you list functions out?

Reply:
i copied this from Lei's code, which also has pipes in the unit tests. I might be able to get to these today.